src/cargo/core/profiles.rs

```diff
@@ -870,6 +925,18 @@ impl fmt::Display for Strip {
     }
 }
 
+impl serde::ser::Serialize for Strip {
+    fn serialize<S>(&self, s: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::ser::Serializer,
+    {
+        match self {
+            Strip::None => "none".serialize(s),
+            Strip::Named(n) => n.serialize(s),
+        }
```

Comment:
```suggestion
        s.collect_str(self)
```

BTW it would be great if we can make commit atomic. Like, this change can fit it one commit of its own.